src/unix/fs.c

```diff
@@ -85,8 +85,11 @@
 #if defined(__CYGWIN__) ||                                                    \
     (defined(__HAIKU__) && B_HAIKU_VERSION < B_HAIKU_VERSION_1_PRE_BETA_5) || \
     (defined(__sun) && !defined(__illumos__)) ||                              \
-    (defined(__APPLE__) && !TARGET_OS_IPHONE &&                               \
-     MAC_OS_X_VERSION_MIN_REQUIRED < 110000)
+    (defined(__APPLE__) &&                                                    \
+      (TARGET_OS_IOS && __IPHONE_OS_VERSION_MIN_REQUIRED < 140000) ||         \
+      (TARGET_OS_TV && __IPHONE_OS_VERSION_MIN_REQUIRED < 140000) ||          \
+      (TARGET_OS_WATCH && __IPHONE_OS_VERSION_MIN_REQUIRED < 70000) ||        \
+      (TARGET_OS_OSX && __MAC_OS_X_VERSION_MIN_REQUIRED < 110000))
```

Comment:
For my edification, why the underscored versions?

Reply:
No definition without underline found except on macOS. I found that it is not possible to accurately judge iOS, TV and watch, maybe checking whether the function exists is a better way.